### ISSUE TITLE
Revisit penalty and loss

### ISSUE THREAD

**Issue by pat-alt** [pat-alt/ECCCo.jl#7]
- [x] set up benchmarking for MNIST
- [x] implement adaptive classifier
- [x] implement ensemble
- [x] #8
- [x] Revisit configurable loss function
- [x] Sort out for classifiers with logit output
